### PR TITLE
Added Croatia countrywide addresses

### LINKED ISSUE #3982: Croatia
URL for the Croatian national register ("Registar prostornih jedinica") is:
https://dgu.gov.hr/proizvodi-i-usluge/registar-prostornih-jedinica-172/172

Prices are defined in official gazette: [NN 148/2008](https://narodne-novine.nn.hr/clanci/sluzbeni/2008_12_148_4082.html) states prices as:
> Popis kućnih brojeva za područje Republike Hrvatske: 6000,00 kn (whole country: 800 EUR / 1000 USD)
> Popis kućnih brojeva za područje jedne županije: 1000,00 kn (each of 21 counties individually: 135 EUR / 166 USD)

Not sure about the license of data purchased this way.

Some data seems to be available from local officials, eg Rijeka #3645.

### PR DESCRIPTION
**Pull request by stefanb**
Fixes #3982

### CODE REVIEW

File: sources/hr/countrywide.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                    "text": "no limitations to public access",
+                    "url": "https://inspire.ec.europa.eu/metadata-codelist/LimitationsOnPublicAccess/noLimitations"
+                },
+                "conform": {
+                    "encoding": "UTF-8",
+                    "format": "xml",
+                    "file": "Address.gml",
+                    "accuracy": 1,
+                    "number": "ad:designator",
+                    "street": "ad:thoroughfareName"
```
Comment (stefanb) [inline sources/hr/countrywide.json:29]: Ran out of ideas how to map these values from GML XML file.... 
Pointers to any known working examples or documentation welcome.

**Comment by iandees** [comment]
I downloaded the data, looked at it with `ogrinfo`, and got an example entry:

```
OGRFeature(Address):2162907451
  gml_id (String) = Address.2162907451
  localId (String) = KB.0000000064
  namespace (String) = HR.DGU.RPJ
  alternativeIdentifier (String) = Bosanska 41 Ada 31000 Osijek
  default (Integer(Boolean)) = 1
  designator (StringList) = (1:41)
  validFrom (String) = 2022-01-31T23:00:00Z
  beginLifespanVersion (String) = 2022-02-01T09:17:58.350Z
  POINT (4999579.43164089 2516702.09078261)
```

This tells me that OGR (which is what we use in our processing pipeline) doesn't know how to pull any of the fields from other namespaces. It looks like OGR needs a file to know how to read these schemas (https://gdal.org/drivers/vector/gml.html#registry-for-gml-application-schemas) and we don't currently have a way to give it that file. It might be easier to download the data as GML and write a script (to put in OpenAddresses' `scripts/` dir) to convert it to something easier to work with. We can host the resulting data if you're up for doing the conversion.

**Comment by stefanb** [comment]
I did manage to make it read `alternativeIdentifier` which is included in the base file and contains the full address (eg "Bosanska 41 Ada 31000 Osijek"), so it may be possible to use that one to parse out the individual fields with regular expressions. Not ideal, but much easier than custom scripts + hosting.

The tricky parts may be addresses with streets containing a number/date within the name, eg "Ulica 1. maja" or such.

**Comment by stefanb** [comment]
Few interesting addresses as regex test cases:
````
Ulica Mirka Miočića hrvatskog erudita i enigmata (1956-2017) 233A Rtina 23248 Ražanac
Fažanska cesta-Valb. 146 Valbandon 52100 Pula (Pola)
Ulica 6. svibnja - Via 6 Maggio 14A Bašanija - Bassanìa 52470 Umag (Umago)
Vinogradska ulica 13/3 Ledinec Gornji 42204 Turčin
Vinogradska ulica 13E Ledinec Gornji 42204 Turčin
Glavani 93/6 Kostrena 51000 Rijeka
Rogocic 11/1 Rogačić 21480 Vis
A-17 A 3 Gajac 53291 Novalja
````

**Comment by mnalis** [comment]
I've been invited here for quick checkup from OpenStreetMap Quick comment; something looks strange  with parsing, e.g. in https://batch.openaddresses.io/job/368812 says:

| hash| number| street |unit | city | district |	region |	postcode |	id |
|-------|-----------|---------|-------|---------|---------|----------|---------------|--------|
37f818a0053d8d03 |	41 |	Bosanska|	Bosanska 41 Ada 31000 Osijek|	Ada	|Osijek|	|	31000|	KB.0000000064|


While https://geoportal.dgu.hr (looking around Osijek) says:

```
PODACI O KUĆNOM BROJU

Kućni broj: 41
Mbr naselja: 19
Naziv naselja: Ada
Identifikator ulice: 190001
Naziv ulice:  Bosanska 
Broj čestice (RPJ): 156/2 
```


I don't know enough about OpenAddresses schema, but in OpenStreetMap (see according to https://wiki.openstreetmap.org/wiki/Tag:boundary=administrative, **grad** / **city**  in Croatia (e.g. [Osijek](https://www.openstreetmap.org/relation/8309777)) would be `admin_level=7`, while **naselje** / **settlement** (e.g. [Ada](https://www.openstreetmap.org/relation/16945302))  would be `admin_level=8`.

**Pull request merged**